ddtrace/appsec/processor.py

```diff
@@ -27,11 +28,41 @@ def get_rules():
     return get_env("appsec", "rules", default=DEFAULT_RULES)
 
 
+COLLECTED_REQUEST_HEADERS = [
+    "accept",
+    "accept-encoding",
+    "accept-language",
+    "content-encoding",
+    "content-language",
+    "content-length",
+    "content-type",
+    "forwarded",
+    "forwarded-for",
+    "host",
+    "true-client-ip",
+    "user-agent",
+    "via",
+    "x-client-ip",
+    "x-cluster-client-ip",
+    "x-forwarded",
+    "x-forwarded-for",
+    "x-real-ip"
+]
+
+
+def _set_headers(span, headers):
+    for k in headers:
+        low = k.lower()
+        if low in COLLECTED_REQUEST_HEADERS:
+            span._set_str_tag(low, headers[k])
```

Comment:
I'd make `COLLECTED_REQUEST_HEADERS` into a set for a faster lookup. And I wonder if we could make it grow dynamically as well so that we can avoid calls to `k.lower()`, that is, if `k` is not in the set, but `k.lower()` is, we add `k` to `COLLECTED_REQUEST_HEADER`

Reply:
sgtm, will change